superset/db_engine_specs.py

```diff
@@ -1185,6 +1193,375 @@ def get_all_datasource_names(cls, db, datasource_type: str) \
                 schema=row['table_schema'], table=row['table_name']))
         return datasource_names
 
+    @classmethod
+    def _build_column_hierarchy(cls,
+                                columns: List[dict],
+                                parent_column_types: List[str],
+                                column_hierarchy: dict) -> None:
+        """
+        Build a graph where the root node represents a column whose data type is in
+        parent_column_types. A node's children represent that column's nested fields
+        :param columns: list of columns
+        :param parent_column_types: list of data types that decide what columns can
+               be root nodes
+        :param column_hierarchy: dictionary representing the graph
+        """
+        if len(columns) == 0:
+            return
+        root = columns.pop(0)
+        root_info = {'type': root['type'], 'children': []}
+        column_hierarchy[root['name']] = root_info
+        while columns:
+            column = columns[0]
+            # If the column name does not start with the root's name,
+            # then this column is not a nested field
+            if not column['name'].startswith('{}.'.format(root['name'])):
```

Comment:
No need for `format` here:

```python
 if not column['name'].startswith(root['name']):
```

Reply:
The period at the end of the root's name is obscure. I needed that period because I ran across a table that had columns like the following:

- ColumnA.nested_obj
- ColumnA.nested_obj1

So without the period, the method would incorrectly assume ColumnA.nested_obj1 was a nested field of ColumnA.nested_obj.

I updated the code to use f strings to perhaps help make the period more obvious.